packages/components/Button/src/Button.styling.ts

```diff
@@ -32,22 +32,60 @@ export const stylingSettings: UseStylingOptions<ButtonProps, ButtonSlotProps, Bu
   tokens: [defaultButtonTokens, defaultButtonFontTokens, defaultButtonColorTokens, buttonName],
   states: buttonStates,
   slotProps: {
-    root: buildProps(
-      (tokens: ButtonTokens, theme: Theme) => ({
-        style: {
-          display: 'flex',
-          alignItems: 'center',
-          flexDirection: 'row',
-          alignSelf: 'flex-start',
-          justifyContent: 'center',
-          width: tokens.width,
-          backgroundColor: tokens.backgroundColor,
-          ...borderStyles.from(tokens, theme),
-          ...layoutStyles.from(tokens, theme),
-        },
-      }),
-      ['backgroundColor', 'width', ...borderStyles.keys, ...layoutStyles.keys],
-    ),
+    ...Platform.select({
+      android: {
+        root: buildProps(
+          (tokens: ButtonTokens) => {
+            return {
+              style: {
+                flexDirection: 'row',
+                alignSelf: 'baseline',
+                backgroundColor: tokens.backgroundColor,
+                borderRadius: tokens.borderRadius,
+                overflow: 'hidden',
+              },
+            };
+          },
+          ['backgroundColor', 'borderRadius'],
+        ),
+        ripple: buildProps(
+          (tokens: ButtonTokens, theme: Theme) => ({
+            style: {
+              display: 'flex',
+              alignItems: 'center',
+              flexDirection: 'row',
+              alignSelf: 'flex-start',
+              justifyContent: 'center',
+              width: tokens.width,
+              ...borderStyles.from(tokens, theme),
+              ...layoutStyles.from(tokens, theme),
+            },
+            android_ripple: {
+              color: tokens.pressed.backgroundColor,
+            },
+          }),
+          ['pressed', 'width', ...layoutStyles.keys, ...borderStyles.keys],
```

Comment:
Not sure if caching will work for `pressed.backgroundColor` as we have here only 'pressed' passed to the array. @rurikoaraki , do you know if this syntax will work?

Reply:
Hm I'm worried about the caching being invalidated more frequently because it would be comparing whether `pressed` has been assigned a new object instead of just a primitive value. Unfortunately the cache won't allow pressed.backgroundColor since that's not part of the ButtonTokens type.

I think for the purposes of making sure the token system works properly, the ripple color should be its own token.